src/main/java/org/shanerx/tradeshop/shop/Shop.java

```diff
@@ -0,0 +1,1211 @@
+		return settingKey.isUsable(shopType) && shopSettings.containsKey(settingKey);
+	}
+
+	/**
+	 * Retrieves the specified setting if it is usable on the shop
+	 */
+	public Map<ShopSettingKeys, ObjectHolder<?>> getShopSettings() {
+		return shopSettings;
+	}
+
+	/**
+	 * Sets the Shops settings based on the passed ShopType
+	 */
+	public void setShopSettings(ShopType type) {
+		if (shopSettings == null)
+			shopSettings = new HashMap<>();
+
+		for (ShopSettingKeys settingKey : ShopSettingKeys.values()) {
+			if (shopSettings.containsKey(settingKey) && !settingKey.isUsable(type)) {
+				shopSettings.remove(settingKey);
+			} else {
+				shopSettings.putIfAbsent(settingKey, settingKey.getDefaultValue(type));
+			}
+
+			if (!settingKey.isUserEditable(type) && shopSettings.get(settingKey) != settingKey.getDefaultValue(type)) {
+				shopSettings.put(settingKey, settingKey.getDefaultValue(type));
+			}
+		}
+	}
+
+	/**
+	 * Retrieves the specified setting if it is usable on the shop
+	 */
+	public void setShopSettings(Map<ShopSettingKeys, ObjectHolder<?>> newSettings) {
+		if (newSettings.size() > 0)
+			new Utils().PLUGIN.getListManager().removeSkippableShop(getShopLocation());
+
+		for (ShopSettingKeys settingKey : newSettings.keySet()) {
+			if (settingKey.isUsable(shopType) && newSettings.get(settingKey) != null) {
+				shopSettings.put(settingKey, newSettings.get(settingKey));
+			}
+		}
+	}
+
+	//------------------------------------------------------------------------------------------------------------------
+	//endregion
+
+	//region User Management - Methods for adding/deleting/updating/viewing a shops users
+	//------------------------------------------------------------------------------------------------------------------
+
+	/**
+	 * Returns a list of all users for the shop based on specified roles.
+	 *
+	 * @param roles role to get users with
+	 * @return list of ShopUsers.
+	 */
+	public List<ShopUser> getUsers(ShopRole... roles) {
+		return getUsersExcluding(Collections.emptyList(), roles);
+	}
+
+	/**
+	 * Returns a list of all users for the shop based on specified roles. Will exclude any users in the list.
+	 *
+	 * @param excludedPlayers player UUIDs to exclude from the results
+	 * @param roles           role to get users with
+	 * @return list of ShopUsers.
+	 */
+	public List<ShopUser> getUsersExcluding(List<UUID> excludedPlayers, ShopRole... roles) {
+		List<ShopUser> users = new ArrayList<>();
+		for (ShopRole role : roles) {
+			switch (role) {
+				case MEMBER:
+					members.forEach(uuid -> {
+						if (!excludedPlayers.contains(uuid)) {
+							users.add(new ShopUser(uuid, role));
+						}
+					});
+					break;
+				case MANAGER:
+					managers.forEach(uuid -> {
+						if (!excludedPlayers.contains(uuid)) {
+							users.add(new ShopUser(uuid, role));
+						}
+					});
+					break;
+				case OWNER:
+					if (!excludedPlayers.contains(owner.getUUID()))
+						users.add(owner);
+					break;
+			}
```

Comment:
too much repetition, would be cool if this could be reduced ( we have the same thing 3 times)

Reply:
could probably use lambda ifs but not sure what else...2.6.1

Reply:
I lied apparently, brought the whole method down to 8 lines w/ no repetition.